mysite/blog/views.py

```diff
@@ -21,3 +24,24 @@ def post_detail(request, year, month, day, post):
         publish__day=day,
     )
     return render(request, "blog/post/detail.html", {"post": post})
+
+
+def post_share(request, post_id):
+    post = get_object_or_404(Post, id=post_id, status="published")
+    if request.method == "POST":
+        form = EmailPostForm(request.POST)
+        if form.is_valid():
+            share_post_by_email(
+                post,
+                form.cleaned_data["name"],
+                form.cleaned_data["comments"],
+                form.cleaned_data["to"],
+            )
+            messages.success(
+                request, f"{post.title} was successfully sent to {form.to}"
+            )
+        else:
+            messages.error(request, "Error sending mail")
+    else:
+        form = EmailPostForm()
+    return render(request, "blog/post/share.html", {"post": post, "form": form})
```

Comment:
These branching conditionals can be simplified to
```suggestion
    if request.method == "POST":
        form = EmailPostForm(request.POST)
        if form.is_valid():
            share_post_by_email(
                post,
                form.cleaned_data["name"],
                form.cleaned_data["comments"],
                form.cleaned_data["to"],
            )
            messages.success(
                request, f"{post.title} was successfully sent to {form.to}"
            )
            return render(request, "blog/post/share.html", {"post": post, "form": form})
            
        messages.error(request, "Error sending mail")
        return render(request, "blog/post/share.html", {"post": post, "form": form})
 
    form =  EmailPostForm()
    return render(request, "blog/post/share.html", {"post": post, "form": form})
```

Too many branching conditionals makes the code difficult to read and complex.